jina/drivers/multimodal.py

```diff
@@ -0,0 +1,72 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from collections import defaultdict
+from typing import Iterable, Tuple
+
+from .encode import BaseEncodeDriver
+from .helper import pb2array, array2pb
+from ..proto import jina_pb2
+
+
+def _extract_doc_content(doc: 'jina_pb2.Document', field: str):
+    if field in ['blob', 'embedding']:
+        return pb2array(getattr(doc, field))
+    elif field:
+        return getattr(doc, field)
+
+
+class MultimodalDriver(BaseEncodeDriver):
+    """Extract multimodal embeddings from different modalities.
+    Input-Output ::
+        Input:
+        document:
+                |- chunk: {modality: mode1}
+                |
+                |- chunk: {modality: mode2}
+        Output:
+        document: (embedding: multimodal encoding)
+                |- chunk: {modality: mode1}
+                |
+                |- chunk: {modality: mode2}
+
+    .. note::
+        - It traverses on the ``documents`` for which we want to apply the ``multimodal`` embedding. This way
+        we can use the `batching` capabilities for the `executor`.
+        - It assumes that every ``chunk`` of a ``document`` belongs to a different modality.
```

Comment:
Look at the documentation here. We should validate this and raise an `Exception` if not fullfilled.